faucet.go

```diff
@@ -818,3 +875,94 @@ func (l *lightningFaucet) generateInvoice(homeTemplate *template.Template,
 		log.Errorf("unable to render home page: %v", err)
 	}
 }
+
+// payInvoice is a hybrid http.Handler that handles: the validation of the
+// pay invoice form, rendering errors to the form, and finally streaming the
+// payment if all the parameters check out.
+func (l *lightningFaucet) payInvoice(homeTemplate *template.Template,
+	homeState *homePageContext, w http.ResponseWriter, r *http.Request) {
+
+	elapsed := time.Since(lastPayInvoiceTime)
+	if elapsed < PayInvoiceTimeout {
+		homeState.SubmissionError = PayInvoiceTimeNotElapsed
+		homeTemplate.Execute(w, homeState)
+		return
+	}
+	lastPayInvoiceTime = time.Now()
+
+	// Get the invoice and try to verify and decode.
+	ctxb := context.Background()
+	payReq := r.FormValue("payinvoice")
+	payReqString := &lnrpc.PayReqString{PayReq: payReq}
+	decodedPayReq, err := l.lnd.DecodePayReq(ctxb, payReqString)
+	if err != nil {
+		log.Errorf("Error on decode pay_req: %v", err)
+		homeState.SubmissionError = ErrorDecodingPayReq
+		homeTemplate.Execute(w, homeState)
+		return
+	}
+
+	decodedAmount := decodedPayReq.GetNumAtoms()
+
+	// Verify invoice amount.
+	if decodedAmount > maxPaymentAtoms {
+		log.Errorf("Max payout, pay_amount: %v", decodedAmount)
+		homeState.SubmissionError = ErrorPaymentAmount
+		homeTemplate.Execute(w, homeState)
+		return
+	}
+
+	// Create the payment request.
+	req := &lnrpc.SendRequest{
+		PaymentRequest:       payReq,
+		Amt:                  decodedAmount,
+		IgnoreMaxOutboundAmt: true,
```

Comment:
Don't ignore max outbound amount, since this means the faucet is out of funds to send payments.